### PR TITLE
Accommodate for the new main branch name of Git for Windows' repositories

### PR DESCRIPTION
**Pull request by dscho**
The main branches were all renamed to `main`, so it is time for us to make sure that everything works in our tooling.

That wasn't too hard: a simple `git grep` followed by manual edits, and clicketyclick, we're done.

Note: there are still a couple of references to the old name, by necessity: for a transitional period of time, the main branch is mirrored to the old name.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The build environment brings all the necessary parts required to build a Git for
 
 ## Git for Windows
 
-The most important part of Git for Windows is [Git](https://git-scm.com/), obviously. The Git for Windows project maintains [a friendly fork](https://github.com/git-for-windows/git) of the "upstream" [Git project](https://github.com/git/git). The idea is that the Git for Windows repository serves as a test bed to develop patches and patch series that are specific to the Windows port, and once the patches stabilized, they are [submitted upstream](https://github.com/git-for-windows/git/tree/master/Documentation/SubmittingPatches).
+The most important part of Git for Windows is [Git](https://git-scm.com/), obviously. The Git for Windows project maintains [a friendly fork](https://github.com/git-for-windows/git) of the "upstream" [Git project](https://github.com/git/git). The idea is that the Git for Windows repository serves as a test bed to develop patches and patch series that are specific to the Windows port, and once the patches stabilized, they are [submitted upstream](https://github.com/git-for-windows/git/tree/main/Documentation/SubmittingPatches).
```
Comment (derrickstolee) [inline README.md:21]: What's different about this line? CRLF?

File: ReleaseNotes.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ Latest update: June 1st 2020
 
 ## Introduction
 
-These release notes describe issues specific to the Git for Windows release. The release notes covering the history of the core git commands can be found [in the Git project](https://github.com/git/git/tree/master/Documentation/RelNotes).
+These release notes describe issues specific to the Git for Windows release. The release notes covering the history of the core git commands can be found [in the Git project](https://github.com/git/git/tree/HEAD/Documentation/RelNotes).
```
Comment (derrickstolee) [inline ReleaseNotes.md:6]: Oooh! I'm going to use this trick when linking to docs in VFS for Git.

File: README.md (hunk 1 above)
Comment (bbolli) [reply README.md:21]: The branch name.

**Comment by bbolli** [comment]
Is there consensus in the wider Git ecosystem for the name `main`?

File: README.md (hunk 1 above)
Comment (dscho) [reply README.md:21]: Yep, for some reason, GitHub does not highlight it. The last URL has `master` in it, and I replaced it by `main`.

Comment (bbolli) [reply README.md:21]: ELINETOOLONG?

**Comment by dscho** [comment]
> Is there consensus in the wider Git ecosystem for the name `main`?

It does look like it, yes. Chrome [announced](https://twitter.com/Una/status/1271180494944829441) that they would rename their main branch, node.js [did the same](https://github.com/nodejs/node/issues/33864), and GitHub, GitLab and BitBucket seem to have indicated that they want to use the name `main` as the default branch name for new repositories at some stage in the near future.

I also submitted [a patch series](https://github.com/gitgitgadget/git/pull/656) to the Git mailing list to make it easier to override, and eventually replace the hard-coded default, for the initial branch after `git init`.

File: README.md (hunk 1 above)
Comment (dscho) [reply README.md:21]: It's an `.md`... so the lines are expected to be wrapped after rendering them.

**Comment by PhilipOakley** [comment]
Minor bikeshed stuff:

It looked like 2/19, 15-17/19 are not about the new main branch change, but other tidy ups that need to coincide with an update at this time. Could/should they be grouped separately?

In 4/19 and 13/19, in the please.sh, the two changes at `1)` and `2)` look like the remove the quote marks, without confirming that the quote mark removal to `main` was deliberate (I think it is?). Also should the two changes be grouped together (It's possibly that they are semantically(?) different patches, but with a part that has a common rationale..)

👍 for the change. I'm happy for `main` to be the branch name on a reference repo. 

Aside: I'm likely to name the main/default branch on my local forks as `mine`, or `duplicate`, because in an ideal world it's just a duplicated (wasted) pointer to `@{upstream}/main`!

[sorry for being late with the comments]

**Comment by dscho** [comment]
> It looked like 2/19, 15-17/19 are not about the new main branch change, but other tidy ups that need to coincide with an update at this time. Could/should they be grouped separately?
> 
> In 4/19 and 13/19, in the please.sh, the two changes at `1)` and `2)` look like the remove the quote marks, without confirming that the quote mark removal to `main` was deliberate (I think it is?). Also should the two changes be grouped together (It's possibly that they are semantically(?) different patches, but with a part that has a common rationale..)

Would you mind commenting on the code? It is a bit tedious to reconstruct what commits and lines you are referring to.

File: README.md (hunk 1 above)
Comment (bbolli) [reply README.md:21]: Yes, I know. I was giving a possible reason why the diff highlighting doesn't work :-)

**Comment by dscho** [comment]
> It looked like 2/19,

The choice I had to make in the `ever-green self-test` fix (2/19) was: do I not even verify that it works after my fixes? I chose not to.

> 15-17/19 are not about the new main branch change, but other tidy ups that need to coincide with an update at this time.

The choice I had to make in 15-17/19 (retiring a bunch of subcommands in `please.sh`) was whether I should adjust the main branch name in those no-longer-relevant subcommands, or whether to retire them immediately and save meself some work. I chose the latter.

> Could/should they be grouped separately?

They could be grouped separately, but as the entire reason for making those changes _now_ was to adjust the relevant scripts for the main branch name, I think it is better to keep them close to exactly those changes.

> In 4/19 and 13/19, in the please.sh, the two changes at `1)` and `2)` look like the remove the quote marks, without confirming that the quote mark removal to `main` was deliberate (I think it is?).

Well, the removal was deliberate, as I am no longer talking about a specific branch _name_, but I talk about The Main Branch (as in: the concept, not the concrete manifestation thereof).

> Also should the two changes be grouped together (It's possibly that they are semantically(?) different patches, but with a part that has a common rationale..)

No longer, as I already merged this to... _consciously reconsiders what they wanted to say..._ the main branch. (There, I almost used the wrong branch name. It will take me as long as everybody else to re-learn.)

**Comment by PhilipOakley** [comment]
Thanks for clarifying. I find the GitHub code commenting method awkward and tend to start with the simple patch view provided in the notification link. Apologies if my overview comment wasn't helpful.

**Comment by dscho** [comment]
@PhilipOakley no worries. Being used to the per-patch review on the Git mailing list, I find the most helpful view to be [this one](https://github.com/git-for-windows/build-extra/pull/296/commits), then clicking on the first commit, commenting, then clicking on the next commit in the drop-down box stating that I look at one specific commit's changes, lather, rinse and repeat.

**Comment by PhilipOakley** [comment]
> I find the most helpful view to be [this one](https://github.com/git-for-windows/build-extra/pull/296/commits), then clicking on the first commit, commenting, then clicking on the next commit in the drop-down box stating that I look at one specific commit's changes, lather, rinse and repeat.

Thanks I'll try that, maybe doubling up with a side screen showing the overall patch view for cross reference.  

When "clicking on the first commit" do you click the message, the three dots (...) or one of the three icons at the right (& does it make any difference..) ?

**Comment by dscho** [comment]
The message. I don't know whether it makes a difference what you click :-)

**Pull request merged**